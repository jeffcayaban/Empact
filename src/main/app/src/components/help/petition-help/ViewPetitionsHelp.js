import React from "react";
import {Col, Grid, Row} from "react-bootstrap";
import PetitionEntry from '../../../other/images/petition-entry.png';
import PetitionPage from '../../../other/images/petition-page.png';
import {Link} from "react-router-dom";
import BackButton from "../../common/BackButton";

/**
 * Displays help information for viewing petitions.
 */

class ViewPetitionsHelp extends React.PureComponent {

    render() {
        return (
            <div className={'secondaryPageContainerStyle'}>
                <div className={"centerStyle"}>
                    <h2>Viewing Petitions</h2>
                    <p>Understand how petitions are viewed.</p>
                </div>
                <div className={"titleContentPadding"}>
                    <Grid className={"divNoPadding"}>
                        <Row>
                            <Col md={12} lg={12}>
                                <div>
                                    <h4 style={{ margin: 0 }}>Viewing petitions on the home, search and explore pages</h4>
                                    <hr className={"helpHr"} />
                                    <p>
                                        You may see petitions on the homepage, search page or the explore page. The
                                        image below explains the different things you may see.
                                    </p>
                                    <div className={"centerStyle"} style={{ paddingTop: '15px' }}>
                                        <img style={{ maxWidth: '60%', minWidth: '350px', align: 'center' }}
                                             src={PetitionEntry}
                                             alt="Petition Entry Diagram" />
                                    </div>
                                </div>
                                <div style={{ paddingTop: '25px' }}>
                                    <h4 style={{ margin: 0 }}>Viewing a petition on its page</h4>
                                    <hr className={"helpHr"} />
                                    <p>
                                        The image below explains the different things you may see when you are on a
                                        petition page.
                                    </p>
                                    <div className={"centerStyle"} style={{ paddingTop: '15px' }}>
                                        <img style={{ maxWidth: '70%', minWidth: '350px', align: 'center' }}
                                             src={PetitionPage}
                                             alt="Petition Page" />
                                    </div>
                                </div>

                                <div style={{ paddingTop: '20px' }}>
                                    <p>
                                        As you scroll down the page, you will see a set of arguments and a button to access
                                        a graph showing the different supporting and opposing arguments that are made to it.
                                        Click <Link to={"/help/argument/view"}>here</Link> to learn more about this section.
                                    </p>
                                </div>
                                <BackButton link={"/help/petition"}/>
                            </Col>
                        </Row>
                    </Grid>
                </div>
            </div>
        );
    }
}

export default ViewPetitionsHelp;