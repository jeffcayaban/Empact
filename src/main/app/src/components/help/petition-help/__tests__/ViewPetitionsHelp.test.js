import React from 'react';
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";
import ViewPetitionsHelp from "../ViewPetitionsHelp";

describe('ViewPetitionsHelp', () => {

    it('should render successfully', () => {
        const component = shallow(<ViewPetitionsHelp />);
        expect(toJson(component)).toMatchSnapshot();
    });
});
