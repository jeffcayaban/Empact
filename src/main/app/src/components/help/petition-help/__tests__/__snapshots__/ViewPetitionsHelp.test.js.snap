// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewPetitionsHelp should render successfully 1`] = `
<div
  className="secondaryPageContainerStyle"
>
  <div
    className="centerStyle"
  >
    <h2>
      Viewing Petitions
    </h2>
    <p>
      Understand how petitions are viewed.
    </p>
  </div>
  <div
    className="titleContentPadding"
  >
    <Grid
      bsClass="container"
      className="divNoPadding"
      componentClass="div"
      fluid={false}
    >
      <Row
        bsClass="row"
        componentClass="div"
      >
        <Col
          bsClass="col"
          componentClass="div"
          lg={12}
          md={12}
        >
          <div>
            <h4
              style={
                Object {
                  "margin": 0,
                }
              }
            >
              Viewing petitions on the home, search and explore pages
            </h4>
            <hr
              className="helpHr"
            />
            <p>
              You may see petitions on the homepage, search page or the explore page. The image below explains the different things you may see.
            </p>
            <div
              className="centerStyle"
              style={
                Object {
                  "paddingTop": "15px",
                }
              }
            >
              <img
                alt="Petition Entry Diagram"
                src="petition-entry.png"
                style={
                  Object {
                    "align": "center",
                    "maxWidth": "60%",
                    "minWidth": "350px",
                  }
                }
              />
            </div>
          </div>
          <div
            style={
              Object {
                "paddingTop": "25px",
              }
            }
          >
            <h4
              style={
                Object {
                  "margin": 0,
                }
              }
            >
              Viewing a petition on its page
            </h4>
            <hr
              className="helpHr"
            />
            <p>
              The image below explains the different things you may see when you are on a petition page.
            </p>
            <div
              className="centerStyle"
              style={
                Object {
                  "paddingTop": "15px",
                }
              }
            >
              <img
                alt="Petition Page"
                src="petition-page.png"
                style={
                  Object {
                    "align": "center",
                    "maxWidth": "70%",
                    "minWidth": "350px",
                  }
                }
              />
            </div>
          </div>
          <div
            style={
              Object {
                "paddingTop": "20px",
              }
            }
          >
            <p>
              As you scroll down the page, you will see a set of arguments and a button to access a graph showing the different supporting and opposing arguments that are made to it. Click 
              <Link
                replace={false}
                to="/help/argument/view"
              >
                here
              </Link>
               to learn more about this section.
            </p>
          </div>
          <BackButton
            link="/help/petition"
          />
        </Col>
      </Row>
    </Grid>
  </div>
</div>
`;
